tasks/add-fbc-contribution/add-fbc-contribution.yaml

```diff
@@ -76,24 +79,43 @@ spec:
         build_tags=$(jq '.fbc.buildTags // []' ${DATA_FILE})
         add_arches=$(jq '.fbc.addArches // []' ${DATA_FILE})
         hotfix=$(jq -r '.fbc.hotfix // "false"' ${DATA_FILE})
+        pre_ga=$(jq -r '.fbc.preGA // "false"' ${DATA_FILE})
         staged_index=$(jq -r '.fbc.stagedIndex // "false"' ${DATA_FILE})
+        product_name=$(jq -r '.fbc.productName // ""' ${DATA_FILE})
+        product_version=$(jq -r '.fbc.productVersion // ""' ${DATA_FILE})
         build_timeout_seconds=$(jq -r --arg build_timeout_seconds ${default_build_timeout_seconds} \
             '.fbc.buildTimeoutSeconds // $build_timeout_seconds' ${DATA_FILE})
-        target_index=$(jq -r '.fbc.targetIndex' ${DATA_FILE})
         fbc_fragment=$(jq -cr '.components[0].containerImage' ${SNAPSHOT_PATH})
 
         timestamp_format=$(jq -r '.fbc.timestampFormat // "%s"' ${DATA_FILE})
         timestamp=$(date "+${timestamp_format}")
 
+        # default index_suffix and targetIndex
+        index_suffix=""
+        target_index=$(params.targetIndex)
+
+        # the targetIndex should be modified based on the hotfix or pre-GA
```

Comment:
I think this could be worded a little more clearly. The next chunk of code is about setting an index_suffix, which is then added to the target_index. Or the comment could stay and the target_index could be overwritten on lines 105 and 111 with lines 115-117 removed

Reply:
What Johnny says makes sense. I think it would be more straightforward if we simply modified the `target_index` variable directly inside the `if`/`elif` branches, rather then saving a separate `tag`/`index_suffix` variable and then if it exists, add it to `target_index`.

Reply:
I rewrote the comment. Take a look if it makes more sense now.

Reply:
Ok. Removed the `index_suffix`.